app/javascript/hyacinth_ui_v1/components/users/UserEdit.js

```diff
@@ -14,35 +13,25 @@ import InputGroup from '../ui/forms/InputGroup';
 import Label from '../ui/forms/Label';
 import FormButtons from '../ui/forms/FormButtons';
 import { Can } from '../../util/ability_context';
+import { GetAuthenticatedUserQuery, GetUserQuery, UpdateUserMutation } from '../../graphql/users';
 import Checkbox from '../ui/forms/inputs/Checkbox';
 
 import BooleanRadioButtons from '../ui/forms/inputs/BooleanRadioButtons';
 
-const GET_USER = gql`
-  query User($id: ID!){
-    user(id: $id) {
-      id
-      firstName
-      lastName
-      email
-      isActive
-      isAdmin
-      permissions
-    }
-  }
-`;
+function UserEdit() {
+  const [authenticatedUser, setAuthenticatedUser] = useState({});
 
-const UPDATE_USER = gql`
-  mutation UpdateUser($input: UpdateUserInput!){
-    updateUser(input: $input){
-      user {
-        id
-      }
-    }
-  }
-`;
+  let gqlResponse = useQuery(
```

Comment:
The authenticated user's abilities are actually available to all components. They are set in the `<App />` component via the AbilityContext. Retrieving the abilities for the logged in user here shouldn't be necessary. 

Abilities can be checked with the `Can` component. I believe it can also be retrieved via a hook. 
```
import { Can } from '../../../util/ability_context';
{ ........... }
<Can I="update" of={{ subjectType: 'Project', stringKey: data.project.stringKey }}>
{ ...... }
</Can>
```

More documentation here: https://github.com/stalniy/casl/tree/master/packages/casl-react